pkg/tsdb/cloudmonitoring/types.go

```diff
@@ -20,67 +20,68 @@ type (
+	grafanaQuery struct {
+		AliasBy         string           `json:"aliasBy"`
+		TimeSeriesList  *timeSeriesList  `json:"timeSeriesList,omitempty"`
+		TimeSeriesQuery *timeSeriesQuery `json:"timeSeriesQuery,omitempty"`
+		// TODO: Merge SloQuery into TimeSeriesList
+		SloQuery *sloQuery `json:"sloQuery,omitempty"`
 	}
 
-	// Used to build MQL queries
-	cloudMonitoringTimeSeriesQuery struct {
-		RefID       string
-		ProjectName string
-		Query       string
-		IntervalMS  int64
-		AliasBy     string
-		timeRange   backend.TimeRange
-		GraphPeriod string
-		logger      log.Logger
+	// These should reflect GCM APIs
+	// timeSeries.list https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.timeSeries/list
+	timeSeriesList struct {
+		ProjectName        string   `json:"projectName"`
+		CrossSeriesReducer string   `json:"crossSeriesReducer"`
+		AlignmentPeriod    string   `json:"alignmentPeriod"`
+		PerSeriesAligner   string   `json:"perSeriesAligner"`
+		GroupBys           []string `json:"groupBys"`
+		Filters            []string `json:"filters"`
+		View               string   `json:"view"`
+		// Not part of the GCM API
+		// TODO: Use API fields instead
+		MetricType   string `json:"metricType"`
+		Preprocessor string `json:"preprocessor"`
 	}
-
-	metricQuery struct {
-		ProjectName        string
-		MetricType         string
-		CrossSeriesReducer string
-		AlignmentPeriod    string
-		PerSeriesAligner   string
-		GroupBys           []string
-		Filters            []string
-		AliasBy            string
-		View               string
-		EditorMode         string
-		Query              string
-		Preprocessor       string
-		PreprocessorType   preprocessorType
-		GraphPeriod        string
+	// TODO: sloQuery can be specified as timeSeriesList paramaters
+	sloQuery struct {
+		SelectorName   string `json:"selectorName"`
+		ServiceId      string `json:"serviceId"`
+		SloId          string `json:"sloId"`
+		LookbackPeriod string `json:"lookbackPeriod"`
 	}
 
-	sloQuery struct {
-		ProjectName      string
-		AlignmentPeriod  string
-		PerSeriesAligner string
-		AliasBy          string
-		SelectorName     string
-		ServiceId        string
-		SloId            string
-		LookbackPeriod   string
+	// timeSeries.query https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.timeSeries/query
+	timeSeriesQuery struct {
+		ProjectName string `json:"projectName"`
+		Query       string `json:"query"`
+		// Not part of the GCM API, will be added to Query
+		GraphPeriod string `json:"graphPeriod"`
 	}
 
-	grafanaQuery struct {
-		DatasourceId int
-		RefId        string
-		QueryType    string
-		MetricQuery  metricQuery
-		SloQuery     sloQuery
-		Type         string
+	// Internal structs. Include computed values
+	// cloudMonitoringTimeSeriesList is used to build time series with a filter
+	cloudMonitoringTimeSeriesList struct {
+		refID   string
+		aliasBy string
+		logger  log.Logger
+		q       *timeSeriesList
```

Comment:
I assume `q` here and below is intended to mean query? I would change this name just for the sake of clarity

Reply:
I'll call this "parameters", the word "query" is a bit overused :sweat_smile: